library/src/main/kotlin/com/connectrpc/protocols/GRPCInterceptor.kt

```diff
@@ -72,30 +72,31 @@ internal class GRPCInterceptor(
                 )
             },
             responseFunction = { response ->
+                val headers = response.headers
                 val trailers = response.trailers
-                val completion = completionParser.parse(trailers)
+                // Handle Trailers-Only response: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses
+                val completion = completionParser.parse(trailers.ifEmpty { headers })
```

Comment:
Fix for trailers-only responses.

Reply:
@akshayjshah mentioned offline that we should be looking for the `grpc-status` in headers _first_, and then in trailers. Additional context: https://github.com/connectrpc/connect-swift/pull/174

Reply:
I'll take a look and alter the algorithm to match.

Reply:
Updated the algorithm to look at headers first, then trailers.